src/java.base/unix/classes/sun/nio/fs/UnixFileAttributeViews.java

```diff
@@ -262,19 +263,43 @@ public UnixFileAttributes readAttributes() throws IOException {
         // chmod
         final void setMode(int mode) throws IOException {
             checkWriteExtended();
-            try {
-                if (followLinks) {
+
+            if (followLinks) {
+                try {
                     chmod(file, mode);
-                } else {
-                    int fd = file.openForAttributeAccess(false);
+                } catch (UnixException e) {
+                    e.rethrowAsIOException(file);
+                }
+                return;
+            }
+
+            if (O_NOFOLLOW == 0) {
+                throw new IOException("NOFOLLOW_LINKS is not supported on this platform");
+            }
+
+            int fd = -1;
+            try {
+                fd = open(file, O_RDONLY, O_NOFOLLOW);
+            } catch (UnixException e1) {
+                if (e1.errno() == EACCES) {
```

Comment:
Maybe we should add a comment like "// retry with write access in case there is no read permission" ?

Reply:
Comment added in 39b0754849d52f5f22e6cc3fb0b48605d58d7f17.